test_communication/test/test_requester.cpp

```diff
@@ -35,14 +36,16 @@ int request(
   size_t number_of_cycles = 5)
 {
   int rc = 0;
-  auto start = std::chrono::steady_clock::now();
-
   auto requester = node->create_client<T>(std::string("test_service_") + service_type);
+  if (!requester->wait_for_service(30_s)) {
```

Comment:
Currently the CTest timeout is 15s (https://github.com/ros2/system_tests/blob/50669368f15265ba3ca1e0c78a8a9086d8247c75/test_communication/CMakeLists.txt#L156). So the 30s in the code don't make much sense and might mislead about what is happening.

Same below.